base/complex.jl

```diff
@@ -457,7 +457,17 @@ function sqrt(z::Complex{<:AbstractFloat})
     end
     Complex(ξ,η)
 end
-sqrt(z::Complex) = sqrt(float(z))
+
+_sqrt(z::Complex{<:AbstractFloat}) = sqrt(z)
+
+function _sqrt(z)
+    ρ = abs(z)
+    θ = angle(z)
+
+    return sqrt(ρ) * cis(θ/2)
```

Comment:
Note that if this were called directly, it would give `_sqrt(-1+0im) == 6.123233995736766e-17 + 1.0im`.  Not sure the large relative error in the real part is a problem for the cases where this fallback would be called?

Reply:
Thanks. I had not understood the original code well enough to just take it over.

The problem is that the `ssqs` function is very specific to IEEE floats. Do you have any suggestions here and in `tanh` for how to avoid overflow without using `ssqs`, or how to make a generic version of that function?

Reply:
For interval arithmetic, the fallback version correctly gives a rigorous bound.